plugins/utils/better-escape.nix

```diff
@@ -0,0 +1,73 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.better-escape;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.better-escape =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "better-escape";
+
+      package = helpers.mkPackageOption "better-escape" pkgs.vimPlugins.better-escape-nvim;
+
+      mapping = helpers.mkNullOrOption (with types; listOf str) ''
+        List of mappings to use to enter escape mode.
+      '';
+
+      timeout =
+        helpers.defaultNullOpts.mkNullable
+        (with types;
+            either ints.unsigned str)
+        "vim.o.timeoutlen" ''
+          The time in which the keys must be hit in ms. Use option timeoutlen by default.
+        '';
+
+      clearEmptyLines = helpers.defaultNullOpts.mkBool false ''
+        Clear line after escaping if there is only whitespace.
+      '';
+
+      keys =
+        helpers.defaultNullOpts.mkNullable
+        (with types;
+            either str helpers.rawType)
+        "<ESC>"
+        ''
+          Keys used for escaping, if it is a function will use the result everytime.
+
+          example(recommended):
+
+          keys = {
+          	_raw = \'\'
+          		function()
+          			return vim.api.nvim_win_get_cursor(0)[2] > 1 and '<esc>l' or '<esc>'
+          		 end,.
+          		 \'\'
+          	};
+        '';
+    };
+
+  config = let
+    setupOptions = with cfg;
+      {
+        inherit mapping keys;
+        timeout =
+          if isInt timeout
```

Comment:
```suggestion
          if isString timeout
```

Reply:
im confused is this is a string shouldnt it be treated as "lua" ?

Reply:
If it is a string, then it should be converted to Lua.
Otherwise (`null` or int), it should be passed as is.